Rakefile

```diff
@@ -1,9 +1,40 @@
 require 'rake'
 require 'rake/testtask'
 
-desc 'rake with no argument will run "rake test"'
-task default: :test
+require_relative 'lib/tasks/exercise_test_task'
 
-Rake::TestTask.new do |task|
-  task.pattern = 'test/**/*_test.rb'
+task default: 'test'
+
+desc 'Run all development and exercise tests'
+task :test do
+  exercises = ARGV.take_while { |e| e != '--' }.drop(1)
+
+  if exercises.any?
+    exercises.each do |e|
+      task(e) { Rake::Task['test:exercise'].execute(exercise: e) }
+    end
+  else
+    Rake::Task['test:dev'].invoke
+    Rake::Task['test:exercises'].invoke
+  end
+end
+
+namespace :test do
+  desc 'Run all development tests located in the test directory'
+  Rake::TestTask.new :dev do |task|
+    task.pattern = 'test/**/*_test.rb'
+  end
+
+  ExerciseTestTask.new :exercise do |task|
+    task.description = "Run the tests for a specific exercise"
+  end
+
+  desc 'Run the tests for all exercises'
+  task :exercises do
+    Dir.foreach('exercises').each do |exercise|
+      next if %w(. ..).include?(exercise)
```

Comment:
This is pretty horrible. Filter out the things you don't want before you get here.